docs/auth/README.adoc

```diff
@@ -13,3 +15,136 @@ dependencies {
 
 == Usage
 
+To use the Auth SDK you'll first need to:
+
+* Import the Core module. See link:../core/README.adoc[the Core documentation].
+* Specify the package name of your Android project in `android.defaultConfig.manifestPlaceholders` of the apps `build.gradle` file.
+This is so the browser needs to know where to redirect to after authentication. See link:https://github.com/aerogear/aerogear-android-sdk/blob/master/example/build.gradle[example app build.gradle].
+* Specify the same URI in the "valid redirect URIs" configuration of the Keycloak Client the app is authenticating with.
+
+=== Retrieving the current user
+
+To retrieve the current authenticated user the `AuthService#currentUser` method can be invoked.
+
+[source,java]
+----
+AuthService authService = mobileCore.getInstance(AuthService.class);
+
+AuthServiceConfiguration authServiceConfig = new AuthServiceConfiguration.AuthConfigurationBuilder()
+    .withRedirectUri("org.aerogear.mobile.example:/callback")
+    .allowSelfSignedCertificate(true)
+    .build();
+
+authService.init(context, authServiceConfig);
+
+UserPrincipal currentUser = authService.currentUser();
+
+if (currentUser) {
+    // User is already authenticated, perform an action
+} else {
+    // User is not authenticates, start authentication flow
+}
+----
+
+=== Authenticating
+
+To start the authentication invoke the `AuthService#login` method.
+
+[source,java]
+----
+static int LOGIN_RESULT_CODE = 1;
+Activity myActivity = this.getActivity();
+
+// Initialise the auth service.
+AuthService authService = (AuthService) mobileCore.getInstance(AuthService.class);
+
+AuthServiceConfiguration authServiceConfig = new AuthServiceConfiguration.AuthConfigurationBuilder()
+    .withRedirectUri("org.aerogear.mobile.example:/callback")
+    .allowSelfSignedCertificate(true)
```

Comment:
same here.

Reply:
actually, let's not init the auth service here again. Let's add a comment here to say at this point, it should be already inited.

Reply:
Moved the initialization out to a separate section. Seems to fit better on its own.